pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -387,8 +411,10 @@ public void recordMessagePersisted(PublishContext publishContext, PositionImpl p
             sequenceId = publishContext.getOriginalSequenceId();
             highestSequenceId = publishContext.getOriginalHighestSequenceId();
         }
-
-        highestSequencedPersisted.put(producerName, Math.max(highestSequenceId, sequenceId));
+        Boolean isLastChunk = (Boolean) publishContext.getProperty(IS_LAST_CHUNK);
```

Comment:
Do we share this IS_LAST_CHUNK context property with other classes or functions? If it is used only in this func, can't we simply define it like this here
`boolean isLastChunk = chunkID != -1 && chunkID == totalChunk - 1;`?

Reply:
We can not get chunk ID and total chunk here.